File: Minimum Length of String After Deleting Similar Ends.cpp
class Solution {
public:
    int minimumLength(string s) {
        int i = 0;
        int j = s.size()-1;

        while (s[i] == s[j] && i != j) {
            char c = s[i];
            i++;
            while (i <= j && s[i] == c) i++;

            j--;
            while (j >= i && s[j] == c) j--;

            if (j < i) return 0;
        }

        return j-i+1;
    }
};

File: Minimum Length of String After Deleting Similar Ends.go
func minimumLength(s string) int {
    i := 0
    j := len(s) - 1

    for s[i] == s[j] && i != j {
        c := s[i]
        i += 1
        for i <= j && s[i] == c {
            i++
        }

        j--
        for j >= i && s[j] == c {
            j -= 1
        }

        if j < i {
            return 0
        }
    }
    return j-i+1
}
